### PR TITLE
Wizard behavior update

### PR DESCRIPTION
**Pull request by martkartasev**
This is a general update to EXISTING behaviors that have had some issues for a long time. New behaviors will be added separately.

1. Major refactoring, renaming and moving of Behavior files.
3. Better separation of concerns in Casting behavior classes.
4. **Fixed charge bug.** As a side effect I had to temporarily disable some of the KeepSafe behavior until I can figure out a better way of doing it.
5. **Improved targeting.** The missile type casting spells will now choose different positions inside the formation for targeting. This is especially useful if the entire formation is not in LoS. In such circumstances each tick to a behavior tries a different trajectory. Each tick essentially causes a random trajectory from the caster to the target formation to be sampled.  If a viable trajectory that has line of sight exists, it is found very quickly (as ticks are fast). Additionally it is also nice for cases where one has multiple casters, so they dont all hit the same position. This will be a fine basis for other offensive casting spells that target enemy formations as well. I will review if any changes need to be made with respect to seeking missiles in the next PR.
6. **Improved formation command behavior**.
   -  **Hold fire** disables all casting.
   -  Wind spell flanking behaviors will not be executed as long as the agent does not have free movement i.e unless it is **skirmishing** under the command of a sergeant (delegate command) or the **charge** command is given.

### CODE REVIEW

**Review by hunharibo — APPROVED** [review]
Very nice changes, no real comment from me.

File: CSharpSourceCode/Battle/AI/AgentBehavior/AgentCastingBehavior/AoEAdjacentCastingBehavior.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        public AoEAdjacentCastingBehavior(Agent agent, AbilityTemplate template, int abilityIndex) : base(agent, template, abilityIndex)
+        {
+            TacticalBehavior = new AdjacentAoETacticalBehavior(agent, agent.GetComponent<WizardAIComponent>(), this);
+        }
+
+        public override void Execute()
+        {
+            var castingPosition = ((AdjacentAoETacticalBehavior) TacticalBehavior)?.CastingPosition;
+            if (castingPosition.HasValue && Agent.Position.AsVec2.Distance(castingPosition.Value.AsVec2) > 5) return;
```
Comment (Badulundabad) [inline CSharpSourceCode/Battle/AI/AgentBehavior/AgentCastingBehavior/AoEAdjacentCastingBehavior.cs:19]: so agent won't cast if distance to target is more than 5?

**Review by Badulundabad — APPROVED** [review]
Great work man, approved. And i have a single one not critical question

File: CSharpSourceCode/Battle/AI/AgentBehavior/AgentCastingBehavior/AoEAdjacentCastingBehavior.cs (hunk 1 above)
Comment (martkartasev) [reply CSharpSourceCode/Battle/AI/AgentBehavior/AgentCastingBehavior/AoEAdjacentCastingBehavior.cs:19]: Kind of. In this case the distance it is from the "casting position" as there is no explicit target. This is specifically for AoE spells that have effects adjacent to the caster. The Wind spell "tactical casting" has a similar behavior but directs the agent to a formation flank instead.

 Other spells can still be cast from further away.

**Pull request merged**